lib/ex_machina.ex

```diff
@@ -96,6 +100,32 @@ defmodule ExMachina do
   end
 
   @doc """
+  Builds a factory with the passed in factory_name and returns its fields
+
+  ## Example
+
+      def factory(:user) do
+        %MyApp.User{name: "John Doe", admin: false}
+      end
+
+      # Returns %{name: "John Doe", admin: true}
+      fields_for(:user, admin: true)
+  """
+  def fields_for(module, factory_name, attrs \\ %{}) do
+    module.build(factory_name, attrs)
+    |> drop_undesireable_fields
+  end
+
+  defp drop_undesireable_fields(record = %{__struct__: struct}) do
+    record
+    |> Map.drop([:__struct__, :__meta__])
+    |> Map.drop(struct.__schema__(:associations))
+  end
+  defp drop_undesireable_fields(record) do
+    record
```

Comment:
Yeah that's a good point actually...so yeah maybe that's the way to think about it. It strips Ecto fields if it's an Ecto Struct and returns a plain map, otherwise. I would be done for that. I guess I was just thinking about it weirdly